### PR TITLE
add howto for registering files

### PR DESCRIPTION
**Pull request by dylanmcreynolds**

### CODE REVIEW

**Comment by danielballan** [comment]
This guide is very short, or the new file was not committed. :-D

**Comment by dylanmcreynolds** [comment]
Hah! Actual file now added.

File: docs/source/how-to/tiled-register.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+## Background
+While Tiled reads files from a file system, it also uses a database to efficiently serve information about the nodes in the tree that it serves. Without a database, tiled must walk the file tree every time it starts up to know about the files that it is serving. 
+
+Tiled has several startup modes, described in [configuration](./configuration) documentation.
+
+### Startup mode: tiled serve directory
+ When Tiled is started up with `tiled serve directory <directory name>`, it takes some time at startup to read the filesystem and store information in its database. For smaller implementations of Tiled, this is very convenient. However, if the file system is large, this process can take too long every time Tiled starts up. 
+
+In this mode, tiled also watches the folder for changes, and reindexes the entire tree when it encounters them. Again, this is fine for smaller instances, but can be too slow if the tree is large. Additionally, watching directories for changes can be inconsistent in cases where the directories are network mounted, as with `NFS` mounted file systems.
```
Comment (Wiebke) [inline docs/source/how-to/tiled-register.md:12]: I believe watching only happens when passing `-w` or `--watch` as an option.

File: docs/source/how-to/tiled-register.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        key_from_filename=identity,
+        path=file_path,
+        prefix=path_prefix,
+        overwrite=False)
+
+
+if __name__ == "__main__":
+    asyncio.run(
+        process_file(
+            # "/tiled_storage/beamlines/8.3.2/recons/rec20240207_120550_test_no_xrays_n257",
```
Comment (Wiebke) [inline docs/source/how-to/tiled-register.md:138]: May want to remove the commented file reference.

**Review by Wiebke — COMMENTED** [review]
This is a super helpful guide for how to ingest files, especially for the use case where Tiled is running and we want to register new data sets through python.

One minor thing: For the sake of completeness, I think it would be worthwhile to include a reference to the possibility of using [`tiled catalog register`](https://blueskyproject.io/tiled/reference/commandline.html#tiled-catalog-register) for a similar, but more granular version of `tiled serve directory`.

File: docs/source/how-to/tiled-register.md (hunk 2 above)
Comment (dylanmcreynolds) [reply docs/source/how-to/tiled-register.md]: removed, thanks

**Comment by danielballan** [comment]
Commits were pulled into #661

**Pull request closed without merging**